common-money.lic

```diff
@@ -48,18 +50,40 @@ module DRCM
   end
 
   def withdraw_exact_amount?(amount_as_string, hometown)
-    bank_id = get_data('town')[hometown]['deposit']['id']
+    settings = get_settings
+    bankbot = settings.bankbot_name
+    bankbot_enabled = settings.bankbot_enabled
+    if bankbot_enabled
+      bank_id = settings.bankbot_room_id
+    else  
+      bank_id = get_data('town')[hometown]['deposit']['id']
+    end  
     DRCT.walk_to(bank_id)
     DRC.release_invisibility
     loop do
-      case DRC.bput("withdraw #{amount_as_string}", 'The clerk counts', 'The clerk tells', 'The clerk glares at you.', 'You count out', 'find a new deposit jar', 'If you value your hands', 'Hey!  Slow down!', "You must be at a bank teller's window to withdraw money")
-      when 'The clerk counts', 'You count out'
-        break true
-      when 'The clerk glares at you.', 'Hey!  Slow down!'
-        pause 15
-      when 'The clerk tells', 'If you value your hands', 'find a new deposit jar', "You must be at a bank teller's window to withdraw money"
-        break false
-      end
+      if bankbot_enabled && DRRoom.pcs.include?(bankbot)
+        amount_convert, type = amount_as_string.split(" ")
+        amount = convert_to_copper(amount_convert, type)
+        currency = hometown_currency(settings.hometown)
+        case DRC.bput("whisper #{bankbot} withdraw #{amount} #{currency}", "offers you", "Whisper what to who?")
+        when "offers you"
+          DRC.bput("accept tip", "You accept")
+          break true
+        when "Whisper what to who?"
+          break false
+        end     
+      else
+        bank_id = get_data('town')[hometown]['deposit']['id']
+        DRCT.walk_to(bank_id)
```

Comment:
This looks redundant, you've already done walk_to bank_id on line 61

Reply:
Not redundant as required if the bankbot is not present. That is what drinfomon is doing, if bankbot_enabled && DRRoom.pcs.include?(bankbot). If the bankbot is not there, it needs to reset bank_id to the hometown bank room number and go there. I do want to move the loop from around the whole block to just the bank part since the bankbot does not care how fast the person seeks money.

Reply:
I'm still not clear on what drinfomon is doing

Reply:
Oh I see its using DRRoom

Reply:
The failsafe behaviour youre talking about never happens though, you dont change `bankbot_enabled` and you break out of the loop when the whisper fails